src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1765,43 +1770,34 @@ private static string FormatErrorMessage(string error, string contentType)
         }
 
         // Returns true if the status code is one of the supported redirection codes.
-        private static bool IsRedirectCode(HttpStatusCode code)
+        private static bool IsRedirectCode(HttpStatusCode statusCode) => statusCode switch
         {
-            int intCode = (int)code;
-            return
-            (
-                (intCode >= 300 && intCode < 304) ||
-                intCode == 307 ||
-                intCode == 308
-            );
-        }
+            HttpStatusCode.Found or
+            HttpStatusCode.Moved or
+            HttpStatusCode.MultipleChoices or
+            HttpStatusCode.PermanentRedirect or
+            HttpStatusCode.SeeOther or
+            HttpStatusCode.TemporaryRedirect => true,
+            _ => false
+        };
 
-        // Returns true if the status code is a redirection code and the action requires switching from POST to GET on redirection.
-        // NOTE: Some of these status codes map to the same underlying value but spelling them out for completeness.
-        private static bool IsRedirectToGet(HttpStatusCode code)
+        // Returns true if the status code is a redirection code and the action requires switching to GET on redirection.
+        // See https://learn.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode
+        private static bool RequestRequiresForceGet(HttpStatusCode statusCode, HttpMethod requestMethod) => statusCode switch
         {
-            return
-            (
-                code == HttpStatusCode.Found ||
-                code == HttpStatusCode.Moved ||
-                code == HttpStatusCode.Redirect ||
-                code == HttpStatusCode.RedirectMethod ||
-                code == HttpStatusCode.SeeOther ||
-                code == HttpStatusCode.Ambiguous ||
-                code == HttpStatusCode.MultipleChoices
-            );
-        }
+            HttpStatusCode.Found or
+            HttpStatusCode.Moved or
+            HttpStatusCode.MultipleChoices => requestMethod == HttpMethod.Post,
+            HttpStatusCode.SeeOther => requestMethod != HttpMethod.Get && requestMethod != HttpMethod.Head,
+            _ => false
+        };
 
         // Returns true if the status code shows a server or client error and MaximumRetryCount > 0
-        private bool ShouldRetry(HttpStatusCode code)
+        private static bool ShouldRetry(HttpStatusCode statusCode) => (int)statusCode switch
         {
-            int intCode = (int)code;
-
-            return
-            (
-                (intCode == 304 || (intCode >= 400 && intCode <= 599)) && WebSession.MaximumRetryCount > 0
-            );
-        }
+            304 or (>= 400 and <= 599) => true,
```

Comment:
The `WebSession.MaximumRetryCount > 0` check is removed, is that by accident?

Reply:
No, i removed it because `ShouldRetry` is called only if `totalRequests > 1` and `totalRequests = WebSession.MaximumRetryCount + 1`, so `WebSession.MaximumRetryCount > 0` is always true